contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -5089,28 +5102,66 @@ transform_pivot_clause(ParseState *pstate, SelectStmt *stmt)
 
 	wrapperSelect_RangeFunction->coldeflist = new_pivot_aliaslist;
 
+	/* make bbf_pivot function argument, default is NULL */
+	bbf_pivot_arg = makeNode(A_Const);
+	bbf_pivot_arg->val.sval.type = T_String;
+	bbf_pivot_arg->val.sval.sval = NULL;
+	bbf_pivot_arg->location = -1;
+	
+	/* 
+	 * There are two ways to handle stmt with pivot operator:
+	 *   1. If the query is wrapped in a view, then we will first genearte a uuid for this pivot call
+	 * 		then two helper views with name like pvt_sv_[uuid], pvt_cv_[uuid]. The uuid will be saved as 
+	 * 		bbf_pivot function argument for later in function view name retrival. 
+	 * 		The uuid, dbid, logical name, view name, and aggreate function name will also be saved in 
+	 *		babelfish_pivot_view catalog.
+	 *   2. If the query is not wrapped in a view, then we will save the pivot operator metadatas
+	 * 		(rawparsetrees, funcName, sql text)to bbf_pivot->context. The metadata will later passed 
+	 * 		to FunctionCallInfo->context and will be consumed within bbf_pivot function.
+	 */
+	is_view_transform = checkQueryWithinView(pstate->p_queryEnv);
+	if (is_view_transform)
+	{	/* case 1: current query is wrapped in a view */
+		char		*view_name;
+		char		*view_uuid;
+		view_name = getQueryEnvViewName(pstate->p_queryEnv);
+		view_uuid = add_entry_to_bbf_pivot_view(view_name,
```

Comment:
are we allowed to use PIVOT in Read-only node? If yes, then this wouldnt work because it does `CatalogInsert`

Reply:
This will also have a side-effect on reader node such that a SELECT + pivot now allocates transaction id. Is this expected and discussed during design reviews?

Reply:
Yes, we can use it. Case 1 only applies to create pivot view. so select pivot will not fall into this scenario. So it will not have side effect on reader.

Reply:
thanks for clarifying!